flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannelTest.java

```diff
@@ -1294,6 +1294,10 @@ static void submitTasksAndWaitForResults(ExecutorService executor, Callable[] ta
 		for (Future result : results) {
 			result.get();
 		}
+
+		// wait until the internal channel state recover task finishes
+		executor.shutdown();
```

Comment:
This util is widely used in many tests and every caller will also shutdown it in finally region. 
So it is better not shutdown it in this common place to avoid this executor has some other usages afterwards.
We can verify the results in finally region after executor shutdown in this specific unit test.